### PR TITLE
GP-781 Check media exists in deep archive

### PR DESCRIPTION
**Pull request by danielhedlund**
## Accessibility

-   [ ] [Tested with screen reader](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#screen-reader)
-   [ ] [Navigable with keyboard](https://github.com/guardian/accessibility/blob/main/people-and-technology/02-physical.md#keyboard)
-   [ ] [Colour contrast passed](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#contrast)
-   [ ] [The change doesn't use only colour to convey meaning](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#use-of-colour)

### CODE REVIEW

File: media_remover/src/main/scala/MediaNotRequiredMessageProcessor.scala [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+
+  def _existsInInternalArchive(onlineOutputMessage: MultiProjectOnlineOutputMessage): Boolean = ???
+
+  def _outputInternalArchiveCopyRequried(onlineOutputMessage: MultiProjectOnlineOutputMessage): Either[String, NearlineRecord] = ???
+
+
+  def processNearlineFileAndProject(vault: Vault, onlineOutputMessage: MultiProjectOnlineOutputMessage, maybeProject: Option[ProjectRecord]):Future[Either[String, MessageProcessorReturnValue]] = {
+    val actionToPerform = maybeProject match {
+      case None =>
+        ("drop_msg", None)
```
[the remark this answers is not in the document]
Comment (fredex42) [reply media_remover/src/main/scala/MediaNotRequiredMessageProcessor.scala:122]: also, might be worth having a log line for each of these cases... just to make it obvious in the logs what the software _thinks_ should be done for a given file.

File: media_remover/src/main/scala/MediaNotRequiredMessageProcessor.scala [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+  def bulkGetProjectMetadata(mediaNotRequiredMsg: MultiProjectOnlineOutputMessage) = {
+    mediaNotRequiredMsg.projectIds.map(id => asLookup.getProjectMetadata(id.toString)).sequence
+  }
+
+  def handleNearline(vault: Vault, mediaNotRequiredMsg: MultiProjectOnlineOutputMessage): Future[Either[String, MessageProcessorReturnValue]] = {
+    /*
+    TODO
+     handle seq of projects
+     if any is new or in production, then silentdrop
+     do we need to handle differing statuses other than that, say if differeing in sensitivity?
```
Comment (fredex42) [inline media_remover/src/main/scala/MediaNotRequiredMessageProcessor.scala:247]: - if status is held and the request is to delete from nearline, then also silentdrop
- other statuses are `Completed` and `Killed`, deletion is appropriate in those circu,stances

**Review by fredex42 — COMMENTED** [review]
Very well done, especially on those test cases. There's still a ways to go but it's coming along really nicely.  Few minor bits and pieces but generally very happy.

File: media_remover/src/main/scala/MediaNotRequiredMessageProcessor.scala [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+        }
+    }
+
+    logger.debug(s"---> actionToPerform: ${actionToPerform._1}")
+
+    performAction(vault, onlineOutputMessage, actionToPerform)
+  }
+
+  private def performAction(vault: Vault, onlineOutputMessage: MultiProjectOnlineOutputMessage, actionToPerform: (String, Option[ProjectRecord])) = {
+    val resultOfAction = actionToPerform match {
```
[the remark this answers is not in the document]
Comment (danielhedlund) [reply media_remover/src/main/scala/MediaNotRequiredMessageProcessor.scala:160]: Ah yes!

**Pull request closed without merging**